src/contracts/services/moneyMarket.ts

```diff
@@ -728,6 +729,104 @@ async function batchQueryMoneyMarketPublicLogs({
   }));
 }
 
+const parseBatchQueryMoneyMarketRewardPools = (responses: any) => (
+  responses.map((response: any) => ({
+    debtMarket: response.id,
+    blockHeight: response.blockHeight,
+    rewardPools: response.response.map((pool: any) => ({
+      rewardPoolId: pool.id,
+      amount: pool.amount,
+      token: pool.token,
+      start: pool.start,
+      end: pool.end,
+      rate: pool.rate,
+    })),
+  }))
+);
+
+function batchQueryMoneyMarketRewardPools$({
+  queryRouterContractAddress,
+  queryRouterCodeHash,
+  lcdEndpoint,
+  chainId,
+  moneyMarket,
+  debtMarkets,
+  batchSize,
+  minBlockHeightValidationOptions,
+  blockHeight,
+}: {
+  queryRouterContractAddress: string,
+  queryRouterCodeHash?: string,
+  lcdEndpoint?: string,
+  chainId?: string,
+  moneyMarket: Contract,
+  debtMarkets: string[],
+  batchSize?: number,
+  minBlockHeightValidationOptions?: MinBlockHeightValidationOptions,
+  blockHeight?: number,
+}) {
+  const queries = debtMarkets.map((debtMarket) => ({
+    id: debtMarket,
+    contract: {
+      address: moneyMarket.address,
+      codeHash: moneyMarket.codeHash,
+    },
+    queryMsg: {
+      reward_pools: {
+        market: debtMarket,
+      },
+    },
+  }));
+  console.error('shadejs reward pools batch queries', queries);
+
+  return batchQuery$({
+    contractAddress: queryRouterContractAddress,
+    codeHash: queryRouterCodeHash,
+    lcdEndpoint,
+    chainId,
+    queries,
+    batchSize,
+    minBlockHeightValidationOptions,
+    blockHeight,
+  }).pipe(
+    map((response) => {
```

Comment:
map(parseBatchQueryMoneyMarketRewardPools) should be good once you take out the logs.